src/features/auth/ui/petInformationForm.tsx

```diff
@@ -234,26 +231,88 @@ const BreedInput = () => {
           <span className="btn-3 text-center text-grey-500">모르겠어요</span>
         </Checkbox>
       </div>
-      <Select isOpen={isOpen} onClose={onClose}>
-        <Select.BottomSheet>
-          <Select.OptionList>
-            {dogBreeds.map((value, idx) => (
-              <Select.Option
-                key={idx}
-                id={value}
-                onClick={() => setBreed(value)}
-                isSelected={value === breed}
-              >
-                {value}
-              </Select.Option>
-            ))}
-          </Select.OptionList>
-        </Select.BottomSheet>
-      </Select>
+      <BreedBottomSheet
+        breed={breed}
+        handleSelect={setBreed}
+        isOpen={isOpen}
+        onClose={() => setIsOpen(false)}
+      />
     </>
   );
 };
 
+interface BreedBottomSheetProps {
+  breed: PetInformationFormExternalState["breed"];
+  handleSelect: (breed: PetInformationFormExternalState["breed"]) => void;
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const BreedBottomSheet = ({
+  breed,
+  handleSelect,
+  onClose,
+  isOpen,
+}: BreedBottomSheetProps) => {
+  const [text, setText] = useState<string>("");
+  const [, startTransition] = useTransition();
+
+  const searchedBreeds = dogBreeds.filter((breed) => {
+    if (text.length === 0) {
+      return true;
+    }
+    const trimedBreed = breed.replace(/\s/g, "");
+    return trimedBreed.includes(text.replace(/\s/g, ""));
+  });
```

Comment:
text 길이가 0이면 바로 dogBreeds를 사용하는 건 어떨까요?

```suggestion
  const searchedBreeds = text.length === 0 ? dogBreeds : dogBreeds.filter((breed) => {
    const trimedBreed = breed.replace(/\s/g, "");
    
    return trimedBreed.includes(text.replace(/\s/g, ""));
  });
```

Reply:
좋네요